src/components/AdminDashboard/AdminDashboard.js

```diff
@@ -70,117 +68,76 @@ const AdminDashboard = () => {
     window.scrollTo(0, 0);
   }, [HAS_VISITED_BEFORE]);
 
+  // resets the selected incidents when switching from approved to unapproved page
+  useEffect(() => {
+    setSelectedIds([]);
+    setAllSelected(false);
+  }, [listType]);
+
   const modalHandler = () => {
     setShowModal(false);
   };
 
-  const lastPage = Math.ceil(unapprovedIncidents.length / incidentsPerPage);
-
   const oktaAxios = useOktaAxios();
 
-  // getting unapproved/pending incidents from the database
-  useEffect(() => {
-    getData(oktaAxios, setUnapprovedIncidents);
-  }, []);
+  const fetchIncidents = () => {
+    getApprovedIncidents(oktaAxios)
+      .then(res => setApprovedIncidents(res))
+      .catch(err => console.log(err));
 
```

Comment:
Hmmm it looks like getData is taking an argument for Unapproved incidents, but in your new function for fetchIncidents, you're getting Approved incidents. Are these functions doing two different things and do we need them both?

Reply:
I am not using the original getData() function anymore (because I hate passing around state setters like that). Three new functions have been added to AdminDashboardHelpers for fetching the three incident lists (approved, pending, form-responses);